src/main/java/be/seeseemelk/mockbukkit/scheduler/BukkitSchedulerMock.java

```diff
@@ -38,11 +43,16 @@ public void shutdown()
 	{
 		waitAsyncTasksFinished();
 		pool.shutdown();
-
+		asyncEventExecutor.shutdownNow();
 		if (asyncException.get() != null)
 			throw new AsyncTaskException(asyncException.get());
 	}
 
+	public Future<?> scheduleAsyncEventCall(Event event)
+	{
```

Comment:
I would suggest adding annotations here as well and even validating the parameter.

```suggestion
	public @NotNull Future<?> scheduleAsyncEventCall(@NotNull Event event)
	{
	    Validate.notNull(event, "Cannot schedule an Event that is null!");
```